src/connectors/configure-related-items/connectConfigureRelatedItems.ts

```diff
@@ -0,0 +1,155 @@
+import algoliasearchHelper, {
+  SearchParameters,
+  PlainSearchParameters,
+} from 'algoliasearch-helper';
+import { Unmounter, WidgetFactory, ResultHit } from '../../types';
+import { createDocumentationMessageGenerator, warning } from '../../lib/utils';
+import connectConfigure, {
+  ConfigureRenderer,
+  ConfigureConnectorParams,
+} from '../configure/connectConfigure';
+
+export type MatchingPatterns = {
+  [attribute: string]: {
+    /**
+     * The score of the optional filter.
+     *
+     * @see https://www.algolia.com/doc/guides/managing-results/rules/merchandising-and-promoting/in-depth/optional-filters/
+     */
+    score: number;
+  };
+};
+
+export interface ConfigureRelatedItemsConnectorParams {
+  /**
+   * The reference hit to extract the filters from.
+   */
+  hit: ResultHit;
+  /**
+   * The schema to create the optional filters.
+   * Each key represents an attribute from the hit.
+   */
+  matchingPatterns: MatchingPatterns;
+  /**
+   * Function called to transform the search parameters generated.
+   */
+  transformSearchParameters?(
+    searchParameters: SearchParameters
+  ): PlainSearchParameters;
+}
+
+type ConfigureConfigureRelatedItemsWidgetFactory<
```

Comment:
It looks like there a typo here and you meant Configure~Configure~RelatedItemsWidgetFactory

Reply:
8a9b9196f2b360d8675e647dc3fadaac46c367f2